### PR TITLE
Initial work to move indentation services down to codestyle layer (step 4/N)

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/60440

This continues that work by pushing ISmartTokenFormatter down to the codestyle layer.

I'm doing these PRs in multiple parts to make things simpler to review. Every time i try to tackle the general problem, it explodes into a ton of changes.

The reason for this change overall is that we have fixers taht need to know indentation information. For example, the 'convert to file-scoped-namespace (and reverse)' fixes have to know indentation information so it can make the right fixes. Not having indentation info be available at thsi level means that the fixes are constrained to the VS host.

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs [hunk 1]
```diff
@@ -63,4 +50,16 @@ protected override IFormattingResult CreateAggregatedFormattingResult(SyntaxNode
         protected override AbstractFormattingResult Format(SyntaxNode node, SyntaxFormattingOptions options, IEnumerable<AbstractFormattingRule> formattingRules, SyntaxToken startToken, SyntaxToken endToken, CancellationToken cancellationToken)
             => new CSharpFormatEngine(node, options, formattingRules, startToken, endToken).Format(cancellationToken);
     }
+
+#if !CODE_STYLE
+    [ExportLanguageService(typeof(ISyntaxFormattingService), LanguageNames.CSharp), Shared]
+    internal class CSharpSyntaxFormattingService : CSharpSyntaxFormatting, ISyntaxFormattingService
```
Comment (mavasani) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs:56]: Can we just move this class directly to Workspaces layer and avoid the `#if !CODE_STYLE`?

File: src/Workspaces/Core/Portable/Indentation/FormattingOptions.IndentStyle.cs [hunk 2]
```diff
@@ -6,6 +6,7 @@ namespace Microsoft.CodeAnalysis.Formatting
 {
     public static partial class FormattingOptions
     {
+        // Publicly exposed.  Keep in sync with IndentStyle2 in the CodeStyle layer.
```
Comment (mavasani) [inline src/Workspaces/Core/Portable/Indentation/FormattingOptions.IndentStyle.cs:9]:
```suggestion
        // Publicly exposed.  Keep in sync with FormattingOptions2.IndentStyle in the CodeStyle layer.
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/FormattingOptions2.IndentStyle.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    internal partial class FormattingOptions2
+    {
+        /// <summary>
+        /// For use in the shared CodeStyle layer.  Keep in syntax with IndentStyle.
```
Comment (mavasani) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/FormattingOptions2.IndentStyle.cs:10]:
```suggestion
        /// For use in the shared CodeStyle layer.  Keep in sync with IndentStyle.
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/ISyntaxFormattingService.cs [hunk 4]
```diff
@@ -17,16 +17,19 @@
...
-#endif
+    internal interface ISyntaxFormatting
     {
         SyntaxFormattingOptions GetFormattingOptions(AnalyzerConfigOptions options);
         IEnumerable<AbstractFormattingRule> GetDefaultFormattingRules();
         IFormattingResult GetFormattingResult(SyntaxNode node, IEnumerable<TextSpan>? spans, SyntaxFormattingOptions options, IEnumerable<AbstractFormattingRule>? rules, CancellationToken cancellationToken);
     }
 
+#if !CODE_STYLE
+    internal interface ISyntaxFormattingService : ISyntaxFormatting, ILanguageService
```
Comment (mavasani) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/ISyntaxFormattingService.cs:28]: Move to Workspaces layer.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Indentation/IndentationOptions.cs [hunk 5]
```diff
@@ -14,6 +14,7 @@ namespace Microsoft.CodeAnalysis.Indentation
         SyntaxFormattingOptions FormattingOptions,
         AutoFormattingOptions AutoFormattingOptions)
     {
```
Comment (mavasani) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Indentation/IndentationOptions.cs:16]: Thinking out loud: Isn't this going to cause a conflict where the same internal type `Microsoft.CodeAnalysis.Indentation.IndentationOptions` is available from both the CodeStyle and Workspaces layer? If you get the build to succeed, then no issues.

File: src/Workspaces/VisualBasic/Portable/Formatting/VisualBasicSyntaxFormattingService.vb [hunk 6]
```diff
@@ -62,4 +46,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Formatting
             Return New VisualBasicFormatEngine(root, options, formattingRules, startToken, endToken).Format(cancellationToken)
         End Function
     End Class
+
+#If Not CODE_STYLE Then
+    <ExportLanguageService(GetType(ISyntaxFormattingService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicSyntaxFormattingService
```
Comment (mavasani) [inline src/Workspaces/VisualBasic/Portable/Formatting/VisualBasicSyntaxFormattingService.vb:52]: Move to VB Workspaces layer.

File: src/Workspaces/VisualBasic/Portable/Microsoft.CodeAnalysis.VisualBasic.Workspaces.vbproj [hunk 7]
```diff
@@ -13,6 +13,10 @@
       .NET Compiler Platform ("Roslyn") support for analyzing Visual Basic projects and solutions.
     </PackageDescription>
   </PropertyGroup>
+  <ItemGroup>
+    <Compile Remove="My Project\**" />
+    <EmbeddedResource Remove="My Project\**" />
+  </ItemGroup>
```
Comment (mavasani) [inline src/Workspaces/VisualBasic/Portable/Microsoft.CodeAnalysis.VisualBasic.Workspaces.vbproj:19]: Unnecessary?

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormattingService.cs [hunk 8]
```diff
@@ -16,11 +16,11 @@
 
 namespace Microsoft.CodeAnalysis.Formatting
 {
-    internal abstract class AbstractSyntaxFormattingService : ISyntaxFormattingService
+    internal abstract class AbstractSyntaxFormatting : ISyntaxFormatting
```
Comment (mavasani) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormattingService.cs:19]: Nit: The file should be renamed to `AbstractSyntaxFormatting.cs`.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs [hunk 9]
```diff
@@ -19,37 +19,24 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Formatting
 {
-#if !CODE_STYLE
-    [ExportLanguageService(typeof(ISyntaxFormattingService), LanguageNames.CSharp), Shared]
-#endif
-    internal class CSharpSyntaxFormattingService : AbstractSyntaxFormattingService
+    internal class CSharpSyntaxFormatting : AbstractSyntaxFormatting
```
Comment (mavasani) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs:22]: Rename the file to `CSharpSyntaxFormatting.cs` and also the VB one to `VisualBasicSyntaxFormatting.vb`

**Review by mavasani — APPROVED** [review]
LGTM, mostly nits.

> Every time i try to tackle the general problem, it explodes into a ton of changes.

I can feel your pain, I ran into the same things when trying to move analyzers/fixers down to these shared layers. Thanks for breaking it up into different PRs!

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs:56]: yes.  that makes sense.  will do in followup!

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/ISyntaxFormattingService.cs (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/ISyntaxFormattingService.cs:28]: yes.  that makes sense.  will do in followup!

File: src/Workspaces/VisualBasic/Portable/Formatting/VisualBasicSyntaxFormattingService.vb (hunk 6 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/VisualBasic/Portable/Formatting/VisualBasicSyntaxFormattingService.vb:52]: yes.  that makes sense.  will do in followup!

File: src/Workspaces/VisualBasic/Portable/Microsoft.CodeAnalysis.VisualBasic.Workspaces.vbproj (hunk 7 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/VisualBasic/Portable/Microsoft.CodeAnalysis.VisualBasic.Workspaces.vbproj:19]: i have no idea why, but these foloders keep coming/going from my projects.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormattingService.cs (hunk 8 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormattingService.cs:19]: yes.  that makes sense.  will do in followup!

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs (hunk 9 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs:22]: yes.  that makes sense.  will do in followup!

**Comment by CyrusNajmabadi** [comment]
> I can feel your pain, I ran into the same things when trying to move analyzers/fixers down to these shared layers. Thanks for breaking it up into different PRs!

yup yup.  i had to do it in tiny bits.  but it all worked out in the end.

File: src/Workspaces/Core/Portable/Indentation/FormattingOptions.IndentStyle.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Indentation/FormattingOptions.IndentStyle.cs:9]: Handled in https://github.com/dotnet/roslyn/pull/60462

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/FormattingOptions2.IndentStyle.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/FormattingOptions2.IndentStyle.cs:10]: Fixes in https://github.com/dotnet/roslyn/pull/60462.  thanks

File: src/Workspaces/VisualBasic/Portable/Formatting/VisualBasicSyntaxFormattingService.vb (hunk 6 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/VisualBasic/Portable/Formatting/VisualBasicSyntaxFormattingService.vb:52]: Done in https://github.com/dotnet/roslyn/pull/60462.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormattingService.cs (hunk 8 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormattingService.cs:19]: DOne in https://github.com/dotnet/roslyn/pull/60462

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs (hunk 9 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs:22]: done in  https://github.com/dotnet/roslyn/pull/60462

**Pull request merged**